chatbot/lib/moderator.js

```diff
@@ -168,32 +172,36 @@ function handleUnBan(user, origin, data) {
 
 /**
  * 
  * @param {string} user 
  * @param {string} message 
  */
-function scanMessage(user, message) {
-    for (let i = 0; i < arrayofUsers.length; i++) {
-        if (arrayofUsers[i][0] == user) {
-            for (let index = 0; index < badWords.length; index++) {
-                if (message.indexOf(badWords[index]) !== -1) {
-                    console.log("aaaaaaaaaaaaaaaaaaaaaaaa")
+function scanMessage(user, message, userID) {
+    if (filterLevel == true) {
+        // The user exists, in the future we would get their rank to determine the filter to check against
+        message = message.split(" ");
+        var badWordFound = false;
+        // for every word...
+        for (let index = 0; index < message.length && badWordFound == false; index++) {
+            // compare against every bad word.
+            for (let badWordArray = 0; badWordArray < badWords.length; badWordArray++) {
```

Comment:
Not very performant? Isn't `badWordArray.indexOf(message[index])` faster?

If not it's still cleaner than looping through every bad word.

Reply:
The problem is indexOf scans the word for a match even if it is just part of a word. Ex. Associate. would trigger the filter because of the first 3 letters. I could use it and make sure the lengths are equal but that would just be another check adding more latency. 

Ideally the check will only be ran if the filter is on and the user has a low rank in the chat. I want to avoid this chain of loops if at all possible. I need to build the rank system first.

Reply:
`.includes()` should do the trick then? Even if it won't be ran much, any latency soon adds up in the end 😁

Reply:
I just tried it, it has the same issue. I think all string comparisons have this problem. Since the minimum action is to timeout the user I really don't want any false positives. 
```js
var str = "Hello world, welcome to the universe.";
var n = str.includes("world");
console.log(n) // true

var str = "Hello worldWITHTEXT, welcome to the universe.";
var n = str.includes("world");
console.log(n) // true
```

It needs to be an exact match and nothing more.

Reply:
Search the other way around! It's true cos it's accurate!

Loop through the message, like you already are, and check if any word on the message is in the bad word array?

Reply:
That works. I switched it to indexOf (scan the array, not the string) and it filtered the message correctly.